src/bin/rational.rs

```diff
@@ -47,6 +49,41 @@ fn mk_constant(n: &BigInt, d: &BigInt) -> Option<Constant> {
     }
 }
 
+fn neg(interval: Interval<Constant>) -> Interval<Constant> {
+    (interval.1.map(|x| x.neg()), interval.0.map(|x| x.neg()))
+}
+
+fn abs(interval: Interval<Constant>) -> Interval<Constant> {
+    let (min, max) = interval;
+    let zero = Ratio::new(0.to_bigint().unwrap(), 1.to_bigint().unwrap());
+
+    let new_min = match (min.clone(), max.clone()) {
+        (None, None) => zero,
```

Comment:
what does `None` encode in these intervals? I thought it was `-inf` and `+inf` depending on which side of the interval it is on. Is that the case? We should add that in a comment if so.

Reply:
not here, but probably in `lib.rs` where Interval is defined.